room.go

```diff
@@ -843,6 +844,39 @@ func (r *Room) handleLocalTrackSubscribed(trackSubscribed *livekit.TrackSubscrib
 	r.callback.OnLocalTrackSubscribed(trackPublication, r.LocalParticipant)
 }
 
+func (r *Room) handleSubscribedQualityUpdate(subscribedQualityUpdate *livekit.SubscribedQualityUpdate) {
+	trackPublication := r.LocalParticipant.getLocalPublication(subscribedQualityUpdate.TrackSid)
+	if trackPublication == nil {
+		r.log.Debugw("recieved subscribed quality update for unknown track", "trackID", subscribedQualityUpdate.TrackSid)
+		return
+	}
+
+	for _, subscribedCodec := range subscribedQualityUpdate.SubscribedCodecs {
+		if !strings.HasSuffix(strings.ToLower(trackPublication.MimeType()), subscribedCodec.Codec) {
+			continue
+		}
+
+		r.log.Infow(
+			"updating subscribed quality",
+			"trackID", trackPublication.SID(),
+			"mime", trackPublication.MimeType(),
+			"subscribedCodec", protoLogger.Proto(subscribedCodec),
+		)
+		for _, subscribedQuality := range subscribedCodec.Qualities {
+			track := trackPublication.GetSimulcastTrack(subscribedQuality.Quality)
+			if track != nil {
+				track.setMuted(!subscribedQuality.Enabled)
+				r.log.Infow(
+					"updating layer enable",
+					"trackID", trackPublication.SID(),
```

Comment:
the two info logs look a bit redundant, if we need both, maybe worth it to make one of them debug.

the "updating subscribed quality" log can be a bit misleading if simulcast track does not have the quality (not sure if there is such a case in the e2e flow, just judging by this code)

Reply:
yeah, I was debating what/where to log this. For the first version, thought of logging both the message and also at layer level just to ensure that they are consistent.

Will change the first log to `handling subscribed quality update` for now.